### PR TITLE
feat: cypress initial commit

### PR DESCRIPTION
**Pull request by mayuran-deriv**
Changes:
I have created two files under 'cypress/home' – one for EU and one for ROW. Since we cannot test both at once.

Screenshot 2023-09-13 at 8 58 09 AM

**success test results -**
Screenshot 2023-09-13 at 8 59 22 AM
Screenshot 2023-09-13 at 8 59 58 AM

 

-   ...

## Type of change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: cypress/e2e/home/home-page.cy.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        // Get all the options within the dropdown
+        cy.get('#radix-6 div').should('have.length', languageArrayLength)
+    })
+
+    it('ROW - Should visit deriv.com on each language', () => {
+        const baseUrl = Cypress.env('base_url') || 'https://deriv.com'
+        Cypress.env('languages').forEach((language) => {
+            const url = `${baseUrl}/${language}/`
+            cy.visit(url)
+            // cy.get('#dm-hero-signup').should('exist')
```
Comment (konstantinosG-deriv) [inline cypress/e2e/home/home-page.cy.ts:27]:
```suggestion
```

File: cypress/e2e/home/home-page-eu.cy.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        // Get all the options within the dropdown
+        cy.get('#radix-6 div').should('have.length', languageArrayLength)
+    })
+
+    it('EU - Should visit eu.deriv.com on each language', () => {
+        const baseUrl = Cypress.env('base_url_eu') || 'https://eu.deriv.com'
+        Cypress.env('languages').forEach((language) => {
+            const url = `${baseUrl}/${language}/`
+            cy.visit(url)
+            // cy.get('#dm-hero-signup').should('exist')
```
Comment (konstantinosG-deriv) [inline cypress/e2e/home/home-page-eu.cy.ts:26]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -157,7 +158,7 @@
         "format": "prettier --write '**/*.js' '*.js'",
         "stylelint": "stylelint 'src/**/*.js' --formatter verbose",
         "eslint": "DEBUG=eslint:cli-engine eslint {'src/**/*.{js,jsx,ts,tsx}','./*.js'} --fix",
-        "start": "npm run develop",
+        "start": "NODE_OPTIONS=--max-old-space-size=16384 npm run develop",
```
[the remark this answers is not in the document]
Comment (mayuran-deriv) [reply package.json:161]: my bad man im gonna remove that

File: cypress.config.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { defineConfig } from 'cypress'
+
+export default defineConfig({
+    e2e: {
+        baseUrl: 'https://deriv.com',
+    },
+    env: {
+        email: 'test@example.com',
+        baseUrl: 'https://deriv.com',
```
Comment (habib-deriv) [inline cypress.config.ts:9]: Let's do the test in staging, then we'll be able to test before code goes to production
```suggestion
        baseUrl: 'https://staging.deriv.com',
```

File: cypress/e2e/home/home-page-eu.cy.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        cy.visit(`${Cypress.env('base_url_eu') || 'https://eu.deriv.com'}`)
+    })
+    it('language popup opens perfectly in eu', () => {
+        cy.get('#radix-4').click()
+    })
+
+    it('EU Language popup Should contain all language items in the dropdown', () => {
+        // Visit the webpage or navigate to the relevant page
+
+        cy.get('#radix-4').click()
```
Comment (habib-deriv) [inline cypress/e2e/home/home-page-eu.cy.ts:13]: Can have the parent tag to make it more readable
```suggestion
        cy.get('header #radix-4').click()
```

File: cypress.config.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        base_url_eu: 'https://eu.deriv.com',
```
Comment (habib-deriv) [inline cypress.config.ts:10]: I just found out we can just test either eu or non-eu url, as for all users it will be redirected anyway. 
```suggestion
```

Comment (mitra-deriv) [reply cypress.config.ts:10]: @habib-deriv Then one test case will always fail? how we will handle it? 🤔

**Pull request closed without merging**